src/libs/get-text-nodes.js

```diff
@@ -0,0 +1,12 @@
+export default el => {
+	const walker = document.createTreeWalker(el, NodeFilter.SHOW_TEXT);
+	const next = () => {
+		const value = walker.nextNode();
+		return {
+			value,
+			done: !value
+		};
+	};
+	walker[Symbol.iterator] = () => ({next});
+	return walker;
+};
```

Comment:
Yeah, it's annoying, but easier if JSDom supports the tree-walker API. See: https://github.com/sindresorhus/is-blob/blob/927899acea40ce7bd31e2b3c4352248eba3a0609/test.js